support/cas-server-support-oauth-api/src/main/java/org/apereo/cas/ticket/accesstoken/AccessToken.java

```diff
@@ -16,6 +16,18 @@
      */
     String PREFIX = "AT";
 
+    /**
```

Comment:
I'd suggest that you remove these changes here, and try to determine JWTness by the ticket id itself.

Reply:
When you first receive an id, you should run the extractor on it before passing it over to the registry and the catalog. If the extractor can parse it as a JWT, you would return the JWT ID which is the original access token id, starting with `AT-`. If the extractor cannot parse it as a JWT, then you simply return it verbatim. Either way, what the extractor returns is the true AT, which can be worked out by the catalog and the registry.

Nowhere else should ever make an assumption about or use ticket prefixes, etc. At, that should be be goal unless 100% required otherwise. That is purely the job of the catalog (and even then, it's still slightly fishy to me TBH and quite likely will be revisited in the future)

Reply:
If I am understanding the design correctly, the flow would be:

1. The extractor will receive an identifier.
2. If it can parse it as a JWT, then the JWT id will be the access token id, which will be AT-<identifier>
3. If it cannot parse it as JWT, then it will use the id verbatim, assuming it's AT-<identifier>

This means that whenever you are processing an access token, you need to call the extractor to locate the AT-<identifier> and then pass that to the registry. The registry will always work with AT-<identifier>. That should work, or else please let me know and I'll run some tests to come up with better suggestions.

Reply:
If that is true, then that is something we should fix first. I'll take a look to confirm this with a few test cases.